/*
Copyright 2017 Google Inc. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS-IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

syntax = "proto2";



package seurat.api.proto;

// 4x4 float matrix.
message Matrix4f {
  // Elements of the matrix in row major order.
  repeated float m = 1 [packed = true];
}

// 2D 32-bit signed integer vector.
message Vector2i {
  // Required. X coordinate
  optional int32 x = 1;

  // Required. Y coordinate
  optional int32 y = 2;
}

// 3D float point.
message Point3f {
  // Required. X coordinate
  optional float x = 1;

  // Required. Y coordinate
  optional float y = 2;

  // Required. Z coordinate
  optional float z = 3;
}
